staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -128,12 +133,27 @@ type OpenAPISchemaInterface interface {
 	OpenAPISchema() (*openapi_v2.Document, error)
 }
 
+type OpenAPIV3Discovery = handler3.OpenAPIV3Discovery
+type OpenAPIV3DiscoveryGroupVersion = handler3.OpenAPIV3DiscoveryGroupVersion
+type OpenAPIV3SchemaInterface interface {
+	OpenAPIV3Discovery() (*OpenAPIV3Discovery, error)
+	OpenAPIV3Schema(schema.GroupVersion) (*openapi_v3.Document, error)
+}
```

Comment:
I'm not a super big fan of this interface, I think something like this would be better (also, generally trying not to use the word discovery which is already overloaded in that area).
```
type OpenAPIv3Root interface {
    Schema(schema.GroupVersion) (*openapi_v3.Document, error)
}

type OpenAPIv3SchemaInterface interface {
    Root() (OpenAPIv3Root, error)
}
```
You can then cache all of these however you want, it also means you don't have to spill a lot of variables inside the DiscoveryClient.

Reply:
@apelisse I am confused about what the `Root()` function provides over the `OpenAPIV3SchemaInterface` implementing `Schema` directly. Additionally, I do not see a way in your proposed interface to list all possible `schema.GroupVersion`s?

Reply:
I notice a possible issue with your proposed naming of `Root` and `Schema`. Given that `Go` uses structural typing, does it make sense for `OpenAPIV3` to take over those non-descriptive function names for any implementers of `OpenAPIv3SchemaInterface`?

Reply:
should be nice now